var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -368,6 +368,21 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     # ###################### Dependencies ##########################
 
+    # External Kokkos
+    depends_on("kokkos@4.1.00", when="@14.4.0: +kokkos")
+    depends_on("kokkos +wrapper", when="trilinos@14.4.0: +kokkos +wrapper")
+    depends_on("kokkos ~wrapper", when="trilinos@14.4.0: +kokkos ~wrapper")
+
+    for a in CudaPackage.cuda_arch_values:
+        arch_str = "+cuda cuda_arch={0}".format(a)
+        kokkos_spec = "kokkos@4.1.00 {0}".format(arch_str)
+        depends_on(kokkos_spec, when="@14.4.0 +kokkos {0}".format(arch_str))
```

Comment:
```suggestion
        arch_str = "+cuda cuda_arch=" + a
        kokkos_spec = "kokkos@4.1.00 " + arch_str
        depends_on(kokkos_spec, when="@14.4.0 +kokkos " + arch_str)
```

Reply:
Why?

Reply:
It's a clearer way of concatenating two strings? Plus `"{0}".format(s)` is only required for python 2.7, so at least change this to `"{}".format(s)`.